### PR TITLE
Add the mongodb extension

### PR DESCRIPTION
**Pull request by alexmace**
This extension supersedes the older mongo extension, as per https://pecl.php.net/package/mongo

### CODE REVIEW

File: manifests/extension/mongodb.pp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  # Get full patch version of PHP
+  $patch_php_version = php_get_patch_version($php)
+
+  # Require php version eg. php::5_4_10
+  # This will compile, install and set up config dirs if not present
+  php_require($patch_php_version)
+
+  $extension = 'mongodb'
+  $package_name = "mongodb-${version}"
+  $url = "http://pecl.php.net/get/mongodb-${version}.tgz"
```
Comment (jacobbednarz) [inline manifests/extension/mongodb.pp]: Can we use a HTTPS endpoint instead?

File: manifests/extension/mongodb.pp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    package_name   => $package_name,
+    package_url    => $url,
+    homebrew_path  => $boxen::config::homebrewdir,
+    phpenv_root    => $php::config::root,
+    php_version    => $patch_php_version,
+    cache_dir      => $php::config::extensioncachedir,
+  }
+
+  # Add config file once extension is installed
+
```
Comment (jacobbednarz) [inline manifests/extension/mongodb.pp]: Don't need this additional newline in here.

File: manifests/extension/mongodb.pp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  require php::config
+
+  # Get full patch version of PHP
+  $patch_php_version = php_get_patch_version($php)
+
+  # Require php version eg. php::5_4_10
+  # This will compile, install and set up config dirs if not present
+  php_require($patch_php_version)
+
+  $extension = 'mongodb'
```
Comment (jacobbednarz) [inline manifests/extension/mongodb.pp]: Would it be possible to line all of the `=` up? Example:

```puppet
$extension    = 'mongodb'
$package_name = "mongodb-${version}"
$url          = "http://pecl.php.net/get/mongodb-${version}.tgz"
$module_path  = "${php::config::root}/versions/${patch_php_version}/modules/${extension}.so"
```

**Comment by alexmace** [comment]
Updated as requested - also made it dependent on Homebrew's openssl, since openssl is required for mongodb

**Comment by jacobbednarz** [comment]
Is this also different from https://github.com/boxen/puppet-php/blob/master/manifests/extension/mongo.pp?

**Comment by alexmace** [comment]
Yes - that's the older version of mongo extension. It was replaced by this one, as per https://pecl.php.net/package/mongo

**Comment by jacobbednarz** [comment]
:cool: let's add some specs and we'll get this merged.

**Comment by alexmace** [comment]
spec added as requested

File: spec/defines/extensions/php_extension_mongodb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+describe "php::extension::mongodb" do
+  #let(:pre_condition) { "class couchbase::lib {}" }
```
Comment (jacobbednarz) [inline spec/defines/extensions/php_extension_mongodb_spec.rb]: if this isn't used, let's remove it.

**Comment by jacobbednarz** [comment]
Thanks a bunch for this! :cake:

**Pull request merged**